### PR TITLE
Fix NPE with unnamed schemas in `{one,any,all}-of`

### LINKED ISSUE #176: Support oneOf without $ref references
Hey folks, I was trying to diff two large OpenAPI descriptions and I could not get it working due to this exception:

```
Unexpected exception. Reason: invalid oneOf schema
```

It's coming from here: https://github.com/OpenAPITools/openapi-diff/blob/master/core/src/main/java/com/qdesrame/openapi/diff/core/compare/schemadiffresult/ComposedSchemaDiffResult.java#L101-L104

I noticed it seems to expect that the schemas under `oneOf` are references, and not an inline schema? For example:

```yaml
schema:
  oneOf:
  - type: string
  - type: object
     properties:
       a:
         type: string
```

Should be valid OpenAPI☝️ But this causes the exception.

### PR DESCRIPTION
**Pull request by gsdatta**
Fixes #176

### CODE REVIEW

File: core/src/main/java/org/openapitools/openapidiff/core/compare/schemadiffresult/ComposedSchemaDiffResult.java [hunk 1]
```diff
@@ -88,10 +90,14 @@ public ComposedSchemaDiffResult(OpenApiDiff openApiDiff) {
...
 
-  private Map<String, Schema> getSchema(Components components, Map<String, String> mapping) {
+  private Map<String, Schema> getSchema(Components components, Map<String, String> mapping, ComposedSchema composedSchema) {
     Map<String, Schema> result = new LinkedHashMap<>();
     mapping.forEach(
         (key, value) -> result.put(key, refPointer.resolveRef(components, new Schema(), value)));
+
+    result.putAll(getUnnamedSchemas(composedSchema.getAllOf(), "all-of"));
+    result.putAll(getUnnamedSchemas(composedSchema.getOneOf(), "one-of"));
+    result.putAll(getUnnamedSchemas(composedSchema.getAnyOf(), "any-of"));
```
Comment (gsdatta) [inline core/src/main/java/org/openapitools/openapidiff/core/compare/schemadiffresult/ComposedSchemaDiffResult.java:100]: quick and dirty workaround - naming the unnamed schemas in order of their appearance in the list.

File: core/src/main/java/org/openapitools/openapidiff/core/compare/schemadiffresult/ComposedSchemaDiffResult.java [hunk 2]
```diff
@@ -119,4 +125,25 @@ public ComposedSchemaDiffResult(OpenApiDiff openApiDiff) {
     return reverseMapping.entrySet().stream()
         .collect(Collectors.toMap(Map.Entry::getValue, Map.Entry::getKey));
   }
+
+  private Map<String, Schema> getUnnamedSchemas(List<Schema> schemas, String name) {
+    Map<String, Schema> result = new LinkedHashMap<>();
+
+    if (schemas == null) {
+      return result;
+    }
```
Comment (gsdatta) [inline core/src/main/java/org/openapitools/openapidiff/core/compare/schemadiffresult/ComposedSchemaDiffResult.java:134]: list is sometimes null (in the diff case) which throws nullptr

**Comment by gsdatta** [comment]
@joschi another fix! this has been throwing a ton of null pointers for us in production so thought I'd put in a quick fix. Would love a review/release!

**Comment by gsdatta** [comment]
Hey @joschi! Any chance we could get a review+release of this? It's been pretty noisy in our diffs, would like to avoid forking if possible 😢

**Review by joschi — APPROVED** [review]
@gsdatta Sorry for the delay. The PR looks good to me. 👍 

Thanks a lot for your contribution!

**Pull request merged**